akka-actor-tests/src/test/scala/akka/routing/MetricsBasedResizerSpec.scala

```diff
@@ -34,29 +41,26 @@ object MetricsBasedResizerSpec {
 
   def routees(num: Int = 10)(implicit system: ActorSystem, timeout: Timeout) = (1 to num).map(_ ⇒ routee).toVector
 
-  case class TestRouter(routees: Vector[ActorRefRoutee], resizer: Resizer)(implicit system: ActorSystem, timeout: Timeout) {
-
-    system.registerOnTermination(close())
+  case class TestRouter(routees: Vector[ActorRefRoutee])(implicit system: ActorSystem, timeout: Timeout) {
 
     var msgs: Set[TestLatch] = Set()
 
-    def mockSend(l: TestLatch = TestLatch(),
-                 routeeIdx: Int = Random.nextInt(routees.length),
-                 wait: Boolean = true)(implicit sender: ActorRef): TestLatch = {
+    def mockSend(await: Boolean,
+                 l: TestLatch = TestLatch(),
+                 routeeIdx: Int = Random.nextInt(routees.length)): Latches = {
       val target = routees(routeeIdx)
-      target.send(l, sender)
+      val first = TestLatch()
```

Comment:
please use more meaningful names, `first` is not clear to me (being first has a purpose, that is more interesting than the fact it being the first).